nodebuilder/tests/fraud_test.go

```diff
@@ -130,20 +129,21 @@ func TestFraudProofSyncing(t *testing.T) {
 
 	require.NoError(t, full.Start(ctx))
 	require.NoError(t, ln.Start(ctx))
-	subsFn, err := full.FraudServ.Subscribe(fraud.BadEncoding)
-	require.NoError(t, err)
-	defer subsFn.Cancel()
-	_, err = subsFn.Proof(ctx)
+	subsFN, err := full.FraudServ.Subscribe(ctx, fraud.BadEncoding)
 	require.NoError(t, err)
 
 	// internal subscription for the fraud proof is done in order to ensure that light node
 	// receives the BEFP.
-	subsLn, err := ln.FraudServ.Subscribe(fraud.BadEncoding)
+	subsLN, err := ln.FraudServ.Subscribe(ctx, fraud.BadEncoding)
 	require.NoError(t, err)
 
-	err = ln.Host.Connect(ctx, *host.InfoFromHost(full.Host))
-	require.NoError(t, err)
```

Comment:
why was this required?

Reply:
i removed it but i still don't understand why it was required. cc @vgonkivs

Reply:
to be 100% sure that full node and light node are connected to each other, so fraud proof can be propagated

Reply:
should we reimplement it then to ensure no flakeyness? @vgonkivs

Reply:
added back